docs/src/components/version-switcher.tsx

```diff
@@ -49,7 +49,7 @@ export default function VersionSwitcher(): JSX.Element {
         mobile={windowSize === 'mobile'}
         items={versions.map(({ label, url }) => ({
           label,
-          to: url + location.pathname + location.hash,
+          to: URL.parse(location.pathname + location.hash, url).href,
```

Comment:
```suggestion
          to: location.href.replace(location.origin, URL.parse(url).origin),
```

If I do this, it handles that issue in case it ever happens again.